### PR TITLE
#293 filtering by tag can produce wrong ids

### LINKED ISSUE #293: v1.2.0-5ab9b86 (Feb 23 2020) gives wrong :id when filtering
I updated to dev and removed my 2 bytes long `tags.data` file. And the first run it got successfully recreated:
```
$ rm ~/.timewarrior/data/tags.data
$ timew summ
Tags database does not exist. Recreating from interval data...

Wk Date       Day Tags                                                                                                        Start      End    Time   Total
W9 2020-02-29 Sat [organze], fix broken TimeWarrior v1.2.0, accidentally tags.data file (JSON) got truncated to zero bytes 20:16:10 20:59:07 0:42:57
                  [.nonwo.], text / browse                                                                                 20:59:07 21:31:25 0:32:18
                  [organze], fix broken TimeWarrior v1.2.0, accidentally tags.data file (JSON) got truncated to zero bytes 21:31:25 22:14:45 0:43:20
                  [.rest......], wc.                                                                                       22:14:46 22:17:15 0:02:29
                  [.rest......], eat / drink                                                                               22:17:16 22:42:12 0:24:56
                  [organze], go reg working time                                                                           22:42:12        - 0:28:06 2:54:06

                                                                                                                                                     2:54:06

$ ll ~/.timewarrior/data/tags.data
-rw-r--r-- 1 saukrs Domain Users 75K Feb 29 23:10 /home/saukrs/.timewarrior/data/tags.data
```

Now if I search for some tag via grep, it gives me right IDs -- `@826` and `@444`:
```
$ timew summ :id :year | grep -- '--- monthly time declaration mark ---'
W1 2020-01-03 Fri @826 --- monthly time declaration mark ---, [pro-time]                                                                                                          10:00:07 10:00:08 0:00:01
                  @444 --- monthly time declaration mark ---, [pro-time]                                                                                                          23:46:52 23:46:56 0:00:04
```

But if I search for it directly via TimeW, ID of the second one, `@825` is falsely shown as equal to the decremented first one, `@826`:
```
$ timew summ :id :year '--- monthly time declaration mark ---'

Wk Date       Day ID   Tags                                                 Start      End    Time   Total
W1 2020-01-03 Fri @826 --- monthly time declaration mark ---, [pro-time] 10:00:07 10:00:08 0:00:01 0:00:01
W5 2020-01-31 Fri @825 --- monthly time declaration mark ---, [pro-time] 23:46:52 23:46:56 0:00:04 0:00:04

                                                                                                   0:00:05
```

Do you know a commit which avoids this bug so I could quickly rollback please?
Ping @sruffell 

---
```
$ timew diag

timew 1.2.0
       Platform: <unknown>

Compiler:
        Version: 7.3.0
           Caps: +stdc +stdc_hosted +200809 +200809 +LP64 +c8 +i32 +l64 +vp64 +time_t64
     Compliance: C++11

Build Features
          Built: Feb 29 2020 20:46:30
         Commit: 5ab9b86
          CMake: 3.13.2
     Build type: release

Configuration
  TIMEWARRIORDB: -
            Cfg: /home/saukrs/.timewarrior/timewarrior.cfg (-rw- 64 bytes)
       Database: /home/saukrs/.timewarrior (drwx 0 bytes)
    Color theme: Built-in default
                 00 01 02 03 04 05 06 07 08 09 10 11 12

Extensions
       Location: /home/saukrs/.timewarrior/extensions (drwx 0 bytes)
               (None)
```

### PR DESCRIPTION
**Pull request by sruffell**
This fixes a bug I introduced in the change where the database is not fully loaded, and adds a test to check for non-contiguous result sets when filtering by tags.

Closes #293

### CODE REVIEW

File: src/data.cpp [hunk 1]
```diff
@@ -625,14 +635,15 @@ std::vector <Range> subtractRanges (
 //
 bool matchesFilter (const Interval& interval, const Interval& filter)
```
[the remark this answers is not in the document]
Comment (sruffell) [reply src/data.cpp]: 👍 ack. Like the previous comment, I'll submit a new pull request in the near future.

File: test/summary.t [hunk 2]
```diff
@@ -137,6 +137,24 @@ W10 2017-03-09 Thu @4 Tag1 8:43:08 9:38:15 0:55:07 0:55:07
                                                    0:55:07
 """, out)
 
+    def test_non_contiguous_with_tag_filter(self):
```
[the remark this answers is not in the document]
Comment (sruffell) [reply test/summary.t]: I see you already have this covered in 41c50e009cd.

[the remark this answers is not in the document]
Comment (lauft) [reply test/summary.t]: SCNR 😄

[the remark this answers is not in the document]
Comment (sruffell) [reply test/summary.t]: :) No worries...actually, I see now that I thought I had put a comment on this issue that I was going to do this but see that I didn't add it.

**Pull request merged**